pkg/internal/storage/repo/utxos.go

```diff
@@ -66,6 +66,17 @@ func (u *UTXOs) CountUTXOs(ctx context.Context, userID int, basketName string) (
 	return count, err
 }
 
+func (u *UTXOs) UnreserveUTXOsByTransactionID(ctx context.Context, transactionID uint) error {
+	err := u.db.WithContext(ctx).Model(&models.UserUTXO{}).
+		Where("reserved_by_id = ?", transactionID).
+		Update("reserved_by_id", nil).Error
+	if err != nil {
+		return fmt.Errorf("failed to unreserve UTXOs by transaction ID %d: %w", transactionID, err)
+	}
+
+	return nil
```

Comment:
maybe use gen query?
```suggestion
	table := u.query.UserUTXO
	_, err := table.WithContext(ctx).
		Where(table.ReservedByID.Eq(transactionID)).
		Update(table.ReservedByID, nil)
	if err != nil {
		return fmt.Errorf("failed to unreserve UTXOs by transaction ID %d: %w", transactionID, err)
	}

	return nil
```